src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -248,6 +263,28 @@ public static ValidationException validateKnnField(
             return exception;
         }
 
+        // Return if vector data type does not need to be checked
+        if (expectedVectorDataType == null) {
```

Comment:
This is the reason why we would like to have overloaded function of validateKnnField , else we will have end up increasing Cyclomatic of this method.

Reply:
Updated